modules/tinymce/src/plugins/tabfocus/main/ts/api/Options.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorOptions } from 'tinymce/core/api/OptionTypes';
+
+const option: {
```

Comment:
I'm afraid with the way everything is setup that's not actually possible without making it a public api as all the plugins are isolated and code can't be shared across for anything in `main` at least.

Reply:
There's been a few things popup like this over time though so maybe we do need to try and solve that, however it's out of scope for now.

Reply:
But we pull in katamari as a separate package why couldn't we have at separate options helpers package. That way you could have common things that then maybe get moved to the public api if they are so common that we think it should live there. But maybe it will be strange if that thing has tinymce types in in. It's just that it feels a bit wrong with all this code duplication that also adds package bloat since the same { processor: 'string', default: <something> } stuff is added hundred of times.

Reply:
I don't really have a problem with this, it's just currying `const option = name => editor => editor.options.get(name)`. It's the types that make it look like a lot of duplication.

Rather than "register string option" I would prefer to have public API processors that are composable. I probably need to think on that some more before I propose anything.

Reply:
> But we pull in katamari as a separate package why couldn't we have at separate options helpers package.

Katamari is entirely different as it's not related to TinyMCE. What we'd need to solve this is a package that's similar to McAgar that is for source content. Now that posses a few problems as we could end up with circular references, so what would be best is to introduce the concept of a `common` module/directory within the TinyMCE where code that is reliant on the editor is shared between plugins, etc... That however does have implications because it's not obvious code you add there is going to bloat when used in plugins.

Anyways, needless to say we had tried to add that common concept once but the idea got shot down. So as I said it's something that needs to be solved outside of this Jira/PR. I also agree with Andy here as the `processor` stuff doesn't worry me as it's no different that repeating `registerString` all the time and then would cause more issues as we add more optional parts to the spec (as I mentioned in a previous PR).

Reply:
One of the reasons I've suggested we investigate code splitting is it will aid library sharing between core and plugins; I have no idea if code splitting will be successful but it seems like the only viable option to solve this concern (we've tried a `common` approach before and it can cause serious maintenance headaches).

Reply:
Right I keep forgetting that the plugins are part of the same package some how I think there is a package boundary between the core and the plugins so yes it would be circular dependencies.

The registerStringOption wouldn't reduce any possible options since if you need fancy things then use register with a record and those fancy things it would just be a short hand for one of the more common things we register.

The repetition is the in type, any code repetition should be avoided if we can. If we wan't to change the type then it's search/replace instead of a common library code change. But I see the circular reference problem now so I understand that it will be hard.

I'm fine with this bloat just wanted to make sure we checked the options here before we convert everything to something that might be hard to switch from later since it's all over the place.